File: server/agents/ReportingAgent.js
const BaseAgent = require('./BaseAgent');

const SYSTEM_PROMPT = `You are the Reporting Agent in an automated incident management system.

You receive the complete incident context and must generate a structured incident report.

INPUT FORMAT (JSON):
{
  "incident_id": "INC-xxx",
  "alert": { <original alert> },
  "detection": { <detection agent output> },
  "decision": { <decision agent output> },
  "action": { <action agent output> },
  "resolution": { <resolution agent output> },
  "timeline": [ { "timestamp": "...", "event": "..." } ],
  "mttd_sec": 12,
  "mttr_sec": 143
}

YOUR TASKS:
Generate a complete incident report with these exact sections:

1. INCIDENT SUMMARY — one paragraph overview
2. TIMELINE — bullet list of events with timestamps
3. ROOT CAUSE ANALYSIS — what caused the issue and why
4. ACTION TAKEN — what was automated and what was manual
5. RESOLUTION — how it was confirmed resolved
6. METRICS — MTTD, MTTR, retry count, automated vs manual
7. RECOMMENDATIONS — 2-3 actionable steps to prevent recurrence

OUTPUT FORMAT (JSON only, no preamble):
{
  "incident_id": "INC-xxx",
  "title": "<short incident title>",
  "summary": "<paragraph>",
  "timeline": ["<timestamp>: <event>"],
  "root_cause": "<explanation>",
  "action_taken": "<what was done>",
  "resolution": "<how confirmed>",
  "metrics": {
    "mttd_sec": 12,
    "mttr_sec": 143,
    "retry_count": 0,
    "automated": true,
    "escalated": false
  },
  "recommendations": ["<rec 1>", "<rec 2>", "<rec 3>"],
  "generated_at": "<ISO timestamp>"
}

RULES:
- Be factual. Only include what is in the input data.
- Output JSON only.`;

class ReportingAgent extends BaseAgent {
  constructor() {
    super('reporting', SYSTEM_PROMPT);
  }

  async run({ incidentId, alert, detection, decision, action, resolution, timeline, mttdSec, mttrSec }) {
    const input = {
      incident_id: incidentId,
      alert: alert || {},
      detection: detection || {},
      decision: decision || {},
      action: action || {},
      resolution: resolution || {},
      timeline: timeline || [],
      mttd_sec: mttdSec || 0,
      mttr_sec: mttrSec || 0,
    };

    return super.run(input);
  }
}

module.exports = new ReportingAgent();

File: incident_management/agents/reporting_agent.py
import logging

from incident_management.agents.base_agent import BaseAgent

logger = logging.getLogger(__name__)

SYSTEM_PROMPT = """You are the Reporting Agent in an automated incident management system.

You receive the complete incident context and must generate a structured incident report.

INPUT FORMAT (JSON):
{
  "incident_id": "INC-xxx",
  "alert": { <original alert> },
  "detection": { <detection output> },
  "decision": { <decision output> },
  "action": { <action output> },
  "resolution": { <resolution output> },
  "timeline": [ { "timestamp": "...", "event": "..." } ],
  "mttd_sec": 12,
  "mttr_sec": 143
}

OUTPUT FORMAT (JSON only, no preamble):
{
  "incident_id": "INC-xxx",
  "title": "<short incident title>",
  "summary": "<paragraph>",
  "timeline": ["<timestamp>: <event>"],
  "root_cause": "<explanation>",
  "action_taken": "<what was done>",
  "resolution": "<how confirmed>",
  "metrics": {
    "mttd_sec": 12,
    "mttr_sec": 143,
    "retry_count": 0,
    "automated": true,
    "escalated": false
  },
  "recommendations": ["<rec 1>", "<rec 2>", "<rec 3>"],
  "generated_at": "<ISO timestamp>"
}

RULES:
- Be factual. Only include what is in the input data.
- Output JSON only."""


class ReportingAgent(BaseAgent):
    def __init__(self) -> None:
        super().__init__("reporting", SYSTEM_PROMPT)

    def run(
        self,
        incident_id: str,
        alert: dict | None = None,
        detection: dict | None = None,
        decision: dict | None = None,
        action: dict | None = None,
        resolution: dict | None = None,
        timeline: list | None = None,
        mttd_sec: int = 0,
        mttr_sec: int = 0,
    ) -> dict:
        payload = {
            "incident_id": incident_id,
            "alert": alert or {},
            "detection": detection or {},
            "decision": decision or {},
            "action": action or {},
            "resolution": resolution or {},
            "timeline": timeline or [],
            "mttd_sec": mttd_sec,
            "mttr_sec": mttr_sec,
        }
        return BaseAgent.run(self, payload)


reporting_agent = ReportingAgent()
